service/SpaceCenter/src/Services/Orbit.cs

```diff
@@ -362,5 +362,138 @@ public double OrbitalSpeedAt (double time)
+        [KRPCMethod]
+        public double RadiusAt(double time)
+        {
+            return InternalOrbit.getRelativePositionAtUT(time).magnitude;
+        }
+
+        /// <summary>
+        /// The orbital radius at the given time, in meters.
+        /// </summary>
+        /// <param name="time">Time from now, in seconds.</param>
+        /// <param name="rf">Reference Frame.</param>
+        [KRPCMethod]
+        public Tuple3 PositionAt(double time, ReferenceFrame rf)
+        {
+            return rf.PositionFromWorldSpace(InternalOrbit.getPositionAtUT(time)).ToTuple();
+
+        }
+
+
+        /// <summary>
+        /// Estimates time of closest approach in the next orbit.
+        /// </summary>
+        /// <param name="target">Other vessel.</param>
+        [KRPCMethod]
+        public double ClosestApproach(Vessel target)
+        {           
+            Orbit b = target.Orbit;
+            double approachTime = Planetarium.GetUniversalTime();
+            double approachDistance = double.MaxValue;
+            double mintime = Planetarium.GetUniversalTime();
+            double interval = InternalOrbit.period;
+            if (InternalOrbit.eccentricity > 1.0) { interval = 100 / InternalOrbit.meanMotion; }
+            double maxtime = mintime + interval;
+
+            //Conduct coarse search
+            double timestep = (maxtime - mintime) / 20;
+            double placeholder = mintime;
+            while (placeholder < maxtime)  
+            {
+                Vector3d PosA = InternalOrbit.getPositionAtUT(placeholder);
+                Vector3d PosB = b.InternalOrbit.getPositionAtUT(placeholder);
+                double thisDistance = Vector3d.Distance(PosA, PosB);
+                if (thisDistance < approachDistance)
+                {
+                    approachDistance = thisDistance;
+                    approachTime = placeholder;
+                }
+                placeholder += timestep;
+            }
+
+            //Conduct fine search
+            double fine_mintime = approachTime - timestep;
+            double fine_maxtime = approachTime + timestep;
+            if (fine_maxtime > maxtime) fine_maxtime = maxtime;
+            if (fine_mintime < mintime) fine_mintime = mintime;
+            timestep = (fine_maxtime - fine_mintime) / 50;
+            placeholder = fine_mintime;
+
+            while (placeholder < fine_maxtime)
+            {
+                Vector3d PosA = InternalOrbit.getPositionAtUT(placeholder);
+                Vector3d PosB = b.InternalOrbit.getPositionAtUT(placeholder);
+                double thisDistance = Vector3d.Distance(PosA, PosB);
+                if (thisDistance < approachDistance)
+                {
+                    approachDistance = thisDistance;
+                    approachTime = placeholder;
+                }
+                placeholder += timestep;
+            }
+            return approachTime;
+        }
+
+
+        /// <summary>
```

Comment:
Any thoughts on how best to organize returning a list of closest approaches?   The options I've come up with are 
-just return a list of times -  users can use the new GetPositionAtUT function to compare vessel and target positions to find the distance themselves
- return a list of two lists - one of times one of distances?   
-return a list of tuples (time, distance)
something smarter you come up with?